uitools/widgets/Esri/ArcGISRuntime/Toolkit/BookmarksView.cpp

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ *  Copyright 2012-2022 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#include "BookmarksView.h"
+
+#include "ui_BookmarksView.h"
+
+// Toolkit headers
+#include "BookmarksViewController.h"
+#include "BookmarkListItem.h"
+
+// ArcGISRuntime headers
+#include <MapGraphicsView.h>
+#include <SceneGraphicsView.h>
+
+namespace Esri {
+namespace ArcGISRuntime {
+namespace Toolkit {
+
+  /*!
+  \class Esri::ArcGISRuntime::Toolkit::BookmarksView
+  \inmodule EsriArcGISRuntimeToolkit
+  \ingroup ArcGISQtToolkitUiCppWidgetsViews
+  \since Esri.ArcGISRuntime 100.15
+  \brief The user interface for the BookmarksView.
+  The BookmarksView displays a collection of bookmarks in the form of viewpoints from either ArcGIS Online, a user-defined portal,
+  or an array of Bookmarks. When the user selects a bookmark from the BookmarksView, the viewpoint rendered in the current
+  geoView is repalced in the given map/scene by the viewpoint selected from the bookmarks.
+
+  \note By default, the BookmarksView will attempt to fetch the set of developer bookmarks, which require an \l{https://developers.arcgis.com/qt/get-started/#3-access-services-and-content-with-an-api-key}{API key} to access.
```

Comment:
❌ World salad, caused by BasemapGallery copy+paste. There's no such thing as developer bookmarks and bookmarks are only part of the Geomodel. 

When a bookmark is selected the geoview's viewpoint is set to the bookmark's stored viewpoint. Avoid mentioning rendering/replacement.